Build fails on FreeBSD: pkg/privileges/privileges.go:6:17: undefined: isPrivileged

### Naabu version:
2.1.7

### Current Behavior:
```
github.com/projectdiscovery/naabu/v2/pkg/privileges
# github.com/projectdiscovery/naabu/v2/pkg/privileges
pkg/privileges/privileges.go:6:17: undefined: isPrivileged
*** Error code 1
```

FreeBSD 13.2

Just ran into the same issue here.

```zsh
vagrant@freebsd13:pts/0-> /home » vagrant (1)
> go version
go version go1.21.3 freebsd/amd64
vagrant@freebsd13:pts/0-> /home » vagrant (0)
> uname -a
FreeBSD freebsd13.localdomain 13.2-RELEASE-p4 FreeBSD 13.2-RELEASE-p4 GENERIC amd64
vagrant@freebsd13:pts/0-> /home » vagrant (0)
> go install -v github.com/projectdiscovery/naabu/v2/cmd/naabu@latest
github.com/projectdiscovery/naabu/v2/pkg/privileges
# github.com/projectdiscovery/naabu/v2/pkg/privileges
Workspace/pkg/mod/github.com/projectdiscovery/naabu/v2@v2.1.9/pkg/privileges/privileges.go:6:17: undefined: isPrivileged
vagrant@freebsd13:pts/0-> /home » vagrant (1)
>
```

--- a/v2/pkg/privileges/privileges_freebsd.go
+++ b/v2/pkg/privileges/privileges_freebsd.go
@@ -1,0 +1,12 @@
+//go:build freebsd
+
+package privileges
+
+import (
+	"os"
+)
+
+// isPrivileged checks if the current process is root
+func isPrivileged() bool {
+	return os.Geteuid() == 0
+}

